storagext/lib/src/clients/market.rs

```diff
@@ -360,6 +373,37 @@ impl MarketClientExt for crate::runtime::client::Client {
             .await
     }
 
+    async fn retrieve_all_deal_parameters(
+        &self,
+    ) -> Result<
+        Vec<(
+            <crate::PolkaStorageConfig as subxt::Config>::AccountId,
+            SpecializedRuntimeDealParameters,
+        )>,
+        subxt::Error,
+    > {
+        let deal_parameters_query = runtime::storage().market().sp_deal_parameters_iter();
+
+        let mut deal_params = self
+            .client
+            .storage()
+            .at_latest()
+            .await?
+            .iter(deal_parameters_query)
+            .await?;
+
+        let mut params = vec![];
+
+        while let Some(Ok(kv)) = deal_params.next().await {
+            let bytes = &kv.key_bytes[(kv.key_bytes.len() - 32)..];
```

Comment:
```suggestion
            let bytes = &kv.key_bytes[..];
```
i don't get why we need `kv.key_bytes.len()`, and there probably is a better way of doing this conversion, but I don't want to bikeshed.

Reply:
Because of this
https://github.com/paritytech/subxt/issues/1201

The API returns the concatenated value with the hash instead of separated

@aidan46 please add the link + an explanation so we don't lose track of this "weird" solution we found